src/powerbi-visual/webpack.config.base.ts

```diff
@@ -0,0 +1,269 @@
+import { BundleAnalyzerPlugin as Visualizer } from 'webpack-bundle-analyzer'
+import MiniCssExtractPlugin from 'mini-css-extract-plugin'
+import { PowerBICustomVisualsWebpackPlugin } from 'powerbi-visuals-webpack-plugin'
+import webpack from 'webpack'
+import fs from 'fs'
+import { WebpackConfiguration } from 'webpack-cli'
+import { VueLoaderPlugin } from 'vue-loader'
+import { TsconfigPathsPlugin } from 'tsconfig-paths-webpack-plugin'
+
+/**
+ * MAIN CONSTS
+ */
+const devServerPort = 8080
+const pbivizPath = './pbiviz.json'
+const capabilitiesPath = './capabilities.json'
+const pluginLocation = './.tmp/precompile/visualPlugin.ts' // path to visual plugin file, the file generates by the plugin
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const powerbiApi: any = powerbi // Types for PowerBI seem to be off, so I'm instead forcing it to `any`
+
+// visual configuration json path
+const pbivizFile = require(path.join(__dirname, pbivizPath))
+
+const packageJsonFile = require(path.join(__dirname, 'package.json'))
+pbivizFile.visual.version = packageJsonFile.version
+
+// the visual capabilities content
+const capabilitiesFile = require(path.join(__dirname, capabilitiesPath))
+
+// string resources
+const resourcesFolder = path.join('.', 'stringResources')
+const localizationFolders = fs.existsSync(resourcesFolder) && fs.readdirSync(resourcesFolder)
+const statsLocation = '../../webpack.statistics.html'
+
+// babel options to support IE11
+const babelOptions = {
+  presets: [
+    [
+      '@babel/preset-env',
+      {
+        targets: {
+          ie: '11'
+        },
+        useBuiltIns: 'entry',
+        corejs: 3,
+        modules: false
+      }
+    ]
+  ],
+  plugins: [],
+  sourceType: 'unambiguous', // tell to babel that the project can contain different module types, not only es2015 modules
+  cacheDirectory: path.join('.tmp', 'babelCache') // path for cache files
+}
+
+export const buildConfig = (params: { mode: 'dev' | 'prod' }) => {
+  const isProd = params.mode === 'prod'
+
+  const loadCert = () => {
+    const keyPath = path.resolve(__dirname, 'localhost-key.pem')
+    const certPath = path.resolve(__dirname, 'localhost.pem')
+    if (!fs.existsSync(keyPath) || !fs.existsSync(certPath)) {
+      console.log('Unable to locate localhost certs, skipping...')
+      return undefined
+    }
+
+    console.log(
+      'Using locally generated localhost certs, make sure the CA cert is installed & trusted!'
+    )
+    return {
+      key: fs.readFileSync(keyPath),
+      cert: fs.readFileSync(certPath)
+    }
+  }
+  const certInfo = isProd ? undefined : loadCert()
+
+  const config: WebpackConfiguration = {
+    entry: {
+      visual: pluginLocation
+    },
+    optimization: {
+      concatenateModules: false,
+      minimize: isProd // enable minimization for create *.pbiviz file less than 2 Mb, can be disabled for dev mode
+    },
+    devtool: isProd ? false : 'inline-source-map',
```

Comment:
1. i'm assuming source maps are unnecessary in the prod build
2. source maps probably didn't work because they were not inlined, which usually would not be an issue, but PowerBI runs all of the JS in a sandbox (w/ `eval()`?) which might not have the same source map support